File: app/src/main/java/com/eericxu/wallpaper/service/Receiver.java
package com.eericxu.wallpaper.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by Eericxu on 2017-08-03.
 */

public class Receiver extends BroadcastReceiver {
	private WallpaperServer server;

	public Receiver(WallpaperServer server) {
		this.server = server;
	}

	@Override
	public void onReceive(Context context, Intent intent) {
		server.onChange();
	}
}

File: app/src/main/java/com/eericxu/wallpaper/service/Receiver.kt
package com.eericxu.wallpaper.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Created by Eericxu on 2017-08-03.
 */
class Receiver(private val server: WallpaperServer) : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        server.onChange()
    }
}
